flow360/component/simulation/models/surface_models.py

```diff
@@ -0,0 +1,128 @@
+"""
+Contains basically only boundary conditons for now. In future we can add new models like 2D equations.
+"""
+
+from abc import ABCMeta
+from typing import List, Literal, Optional, Tuple, Union
+
+import pydantic as pd
+
+from flow360.component.simulation.framework.base_model import Flow360BaseModel
+
+from .turbulence_quantities import TurbulenceQuantitiesType
+
+BoundaryVelocityType = Unioin[VelocityType.Vector, Tuple[pd.StrictStr, pd.StrictStr, pd.StrictStr]]
+
+
+class BoundaryBase(Flow360BaseModel, metaclass=ABCMeta):
+    type: str = pd.Field()
+    surfaces: EntityList[Surface] = pd.Field()
+
+
+class BoundaryBaseWithTurbulenceQuantities(BoundaryBase, metaclass=ABCMeta):
+    turbulence_quantities: Optional[TurbulenceQuantitiesType] = pd.Field(None)
+
+
+class Wall(BoundaryBase):
+    """Replace Flow360Param:
+    - NoSlipWall
+    - IsothermalWall
+    - HeatFluxWall
+    - WallFunction
+    - SolidIsothermalWall
+    - SolidAdiabaticWall
+    """
+
+    model_config = ConfigDict(
+        conflicting_fields=[Conflicts("temperature", "heat_flux")],
+    )
+
+    type: Literal["Wall"] = pd.Field("Wall", frozen=True)
+    use_wall_function: bool = pd.Field(False)
+    velocity: Optional[BoundaryVelocityType] = pd.Field(None)
+    velocity_type: Literal["absolute", "relative"] = pd.Field("relative")
+    temperature: Optional[Union[pd.PositiveFloat, pd.StrictStr]] = pd.Field()
```

Comment:
Let's just copy the Flow360Param unit type here. Unit system should be ready soon.